src/basic/log.c

```diff
@@ -1341,6 +1344,11 @@ void log_received_signal(int level, const struct signalfd_siginfo *si) {
                          signal_to_string(si->ssi_signo));
 }
 
+void set_log_syntax_callback(log_syntax_callback_t cb, void *userdata) {
```

Comment:
maybe we should add an assert() that if the pointers are already set non-NULL we abort. After all, if there are multiple users of this, something is wrong, the stuff is not designed for that